Mage.Sets/src/mage/cards/g/GluttonousHellkite.java

```diff
@@ -0,0 +1,167 @@
+        this.toughness = new MageInt(3);
+
+        // When you cast this spell, each player sacrifices X creatures. Gluttonous Hellkite enters the battlefield with two +1/+1 counters on it for each creature sacrificed this way.
+        Ability ability = new CastSourceTriggeredAbility(new GluttonousHellkiteEffect());
+        ability.addEffect(new GluttonousHellkiteReplacementEffect());
+
+        this.addAbility(ability);
+
+        // Flying
+        this.addAbility(FlyingAbility.getInstance());
+
+        // Trample
+        this.addAbility(TrampleAbility.getInstance());
+
+    }
+
+    private GluttonousHellkite(final GluttonousHellkite card) {
+        super(card);
+    }
+
+    @Override
+    public GluttonousHellkite copy() {
+        return new GluttonousHellkite(this);
+    }
+}
+
+class GluttonousHellkiteEffect extends OneShotEffect {
+
+    DynamicValue xValue;
+
+    GluttonousHellkiteEffect() {
+        super(Outcome.Sacrifice);
+        this.staticText = "each player sacrifices X creatures";
+    }
+
+    private GluttonousHellkiteEffect(final GluttonousHellkiteEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public GluttonousHellkiteEffect copy() {
+        return new GluttonousHellkiteEffect(this);
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+
+        List<Permanent> sacPermanents = new ArrayList<>();
+        Player controller = game.getPlayer(source.getControllerId());
+        Object obj = getValue(CastSourceTriggeredAbility.SOURCE_CAST_SPELL_ABILITY);
+        int sacrificeAmount = 0;
+        if (controller == null ) {
+            return false;
+        }
+        if (obj instanceof SpellAbility) {
+            sacrificeAmount = ((SpellAbility) obj).getManaCostsToPay().getX();
+        }
+        if (sacrificeAmount < 1) {
+            return true;
+        }
+        for (UUID playerId : game.getState().getPlayersInRange(controller.getId(), game)) {
+            Player player = game.getPlayer(playerId);
+            if (player != null) {
+                TargetSacrifice target = new TargetSacrifice(sacrificeAmount, StaticFilters.FILTER_PERMANENT_CREATURE);
+                if (target.canChoose(player.getId(), source, game)) {
+                    player.choose(Outcome.Sacrifice, target, source, game);
+                    target.getTargets().stream().map(game::getPermanent).filter(Objects::nonNull).forEach(sacPermanents::add);
+                }
+            }
+        }
+        int counters = sacPermanents
+                .stream()
+                .mapToInt(p -> p.sacrifice(source, game) ? 2 : 0)
+                .sum();
+
+
+        for (Effect effect : source.getEffects()) {
+            if (effect instanceof GluttonousHellkiteReplacementEffect) {
+                effect.setValue("GluttonousHellkiteCounters", counters);
+            }
+        }
+        return true;
+    }
+}
+
+class GluttonousHellkiteReplacementEffect extends ReplacementEffectImpl {
```

Comment:
Must use standard code with dynamic value. Do not copy paste such complex/global effects. Example:

![IMG_0561](https://github.com/magefree/mage/assets/8344157/720fb8d8-259c-46b4-8621-ada5b88f19e8)

Reply:
How would you go about creating a DynamicValue for the amount of creatures sacrificed with the cast ability?

Reply:
simplest way - just set the number you need as "damage" to all the effects of the source and use SavedDamageValue (yes I know it isn't damage, but it's still a saved value, and it won't interfere with anything)

Reply:
but also, I'm pretty sure this does need to be a custom effect, because it's a cast source trigger and not an EntersBattlefieldAbility